### ISSUE TITLE
[java] ConsecutiveLiteralAppend false-positive with builder inside lambda

### ISSUE THREAD

**Issue by winhkey** [pmd/pmd#2427]
**Affects PMD Version:** 
6.22.0
**Description:**
ConsecutiveLiteralAppends
https://pmd.github.io/latest/pmd_rules_java_performance.html#consecutiveliteralappends

```java
        consumerMap.put(Collection.class, o ->
        {
            builder.append('[');
            Collection<?> collection = (Collection<?>) o;
            if (isNotEmpty(collection))
            {
                collection.forEach(t -> buildValue(builder, t));
                builder.delete(builder.lastIndexOf(", "), builder.length());
            }
            builder.append(']'); // here reported
        });

        consumerMap.put(Map.class, o ->
        {
            builder.append('{');
            Map<?, ?> map = (Map<?, ?>) o;
            if (isNotEmpty(map))
            {
                map.forEach((key, value) ->
                {
                    builder.append(key).append('=');
                    buildValue(builder, value);
                });
                builder.delete(builder.lastIndexOf(", "), builder.length());
            }
            builder.append('}'); // but here is ok
        });
```

**Comment by winhkey** [comment]
I think it has nothing to do with lambda, if I replace lambda with for loop, the error will still comes.

**Comment by adangel** [comment]
Thanks for checking. Could you provide a simplified code example?

**Comment by neowu** [comment]
I encountered similar issues, with PMD 6.31.0/6.32.0

the following code are caused both PMD.InsufficientStringBufferDeclaration, PMD.ConsecutiveLiteralAppends
PMD 6.30.0 works fine

```
@SuppressWarnings({"PMD.InsufficientStringBufferDeclaration", "PMD.ConsecutiveLiteralAppends"})    // pmd false positive
    static String escapeHTML(String text) {
        int length = text.length();
        int index = findHTMLReservedChar(text);
        if (index == length) return text;
        var builder = new StringBuilder(length * 2);
        for (int i = 0; i < index; i++) builder.append(text.charAt(i));
        for (; index < length; index++) {
            char ch = text.charAt(index);
            switch (ch) {
                case '<' -> builder.append("&lt;");
                case '>' -> builder.append("&gt;");
                case '"' -> builder.append("&quot;");
                case '&' -> builder.append("&amp;");
                case '\'' -> builder.append("&#39;");
                case '/' -> builder.append("&#47;");
                default -> builder.append(ch);
            }
        }
        return builder.toString();
    }
```

**Comment by adangel** [comment]
@neowu Thanks for reporting. Your case is a different one, hence I created a separate issue for it -> #3152 

The original issue is fixed since PMD 6.25.0. I'll add a test case for it.
